[#87] Feat: 짤 이미지 신고 확인 모달 UI 구현

HY219 opened the pull request:
## 📝 작업 내용

- 짤 이미지 신고 확인 모달 UI 구현
- 짤 이미지 상세 페이지에서 신고 아이콘 클릭 시 모달 열기

## 📷 스크린샷 (선택)

## 💬 리뷰 요구사항(선택)
- 확인 후 리뷰 부탁드립니다:)
- API 관련 코드 없이, 모달 UI 만 구현했는데, 아래 문제가 발생합니다. 모든 PR에 발생하고 있는데, 다른 분들은 문제 없으신가요?
![image](https://github.com/zzalmyu/zzalmyu-frontend/assets/82134668/cf9fc65f-cff1-45e9-a079-d39a6c2f41c1)
  - 정확한 원인은 찾지 못했고, 찾아본 내용에는 `권한 문제`, `캐싱 문제`, `배포 ID를 자신의 ID로 교체하지 않은 문제` 가 있었습니다.
  배포 설정이나 deploy.yml 파일을 다시 살펴봐야 할 것 같은데 도움 요청합니다!
  - https://stackoverflow.com/questions/72262062/trying-to-add-a-cloudfront-invalidation-command-in-buildspec-yml-throws-254-erro
  - https://github.com/actions/setup-node/issues/516
  - https://github.com/ahanoff/how-to/blob/b1bfa8f32a3862533a5db2aac0405b8b45737598/docusaurus-3-deployment-with-s3-and-cloudfront/README.md?plain=1


## 📍 기타 (선택)

- UI 확인 시 `./src/routes/_layout-with-chat/index.tsx` 파일의 7번째 줄에 다음 코드를 추가해주세요.
  >  <ImageDetailModal isOpen={true} onClose={() => {}} />
- 신고된 사진 제거 확인 모달 UI 코드 참고하여 구현
- TODO: 짤 이미지 신고 API 구현 코드와 연결 및 API 요청 성공/실패 시 동작 구현

close #87

HY219 committed:
Feat: 짤 이미지 신고 확인 모달 UI 구현

신고된 사진 제거 확인 모달 UI 코드 참고하여 구현
TODO: 짤 이미지 신고 api 연결 및 성공/실패 시 동작 구현

thalassophilia commented:
@HY219 서버의 cloudFront 배포 id가 변경됐거나 수정사항이 있는 것 같아요! 오늘 아침에 확인해보겠습니다!

thalassophilia commented:
@HY219 cli로 확인해본 결과, 백엔드에서 넘겨준 cloudFront id가 유효하지 않네요
<img width="674" alt="스크린샷 2024-03-04 오전 4 31 13" src="https://github.com/zzalmyu/zzalmyu-frontend/assets/61978339/9938f108-bcc0-4c2f-a158-5607c4c9ea0e">

thalassophilia commented on src/components/ReportConfirmModal/index.tsx:
```diff
@@ -0,0 +1,50 @@
+  };
+  return (
+    <Modal isOpen={isOpen} onClose={onClose} className="p-40pxr">
+      <Modal.Header hasCloseButton={true}> </Modal.Header>
```
이렇게 해주셔도 될 것 같아요~
```js
<Modal.Header hasCloseButton />
```

HY219 replied:
의견 주신 것이 간결해서 좋은 것 같아요 👍 

다만, `hasCloseButton = {true}` 처럼 true를 명시적으로 작성해주는 것이 좀 더 분명할 것 같은데 어떻게 생각하시나요?

thalassophilia replied:
사실 이거는 정답이 없는 취향의 영역이기도 해요! 혜연님께서 생각하시기에 명시하는 쪽이 더 분명하다고 느끼시면 이렇게 가도 될 거 같아요, 이 부분은 아예 컨벤션으로 정해도 좋을 것 같습니다!
다른 분들은 어떻게 생각하시나요? 의견 들어보고 다수 의견으로 컨벤션 정할게요~~ @HY219 @yjc2021 @choi-ik @hyeonjinan096

hyeonjinan096 replied:
둘 다 좋은데 개인적으론 <Modal.Header hasCloseButton /> 좋은 것 같습니다.
hasCloseButton이 기본적으로 false여서 사용하지 않을 땐 <Modal.Header /> 이렇게 하게 되니까 boolean 값을 명시하지 않아도 되지 않을까 합니다:)

yjc2021 replied:
저도 동적으로 값이 변하는 게 아니라 애초에 고정된 값을 개발자가 주입하는 경우라면 
```ts
<Modal.Header hasCloseButton />
```
을 선호합니다 :)

HY219 replied:
의견주신대로 hasCloseButton에 주었던 true 값을 제거했습니다! props가 사용될 때 항상 같은 값을 사용하는 경우 의견주신 것처럼 작성하는 것 좋은 방법인 것 같습니다! 배워갑니다 😀

thalassophilia replied:
@HY219 네넵 두 방법 다 틀린게 아니고 현업에서는 true, false 명시하는 걸 더 선호하기도 하니 참고만 해두세요!!

thalassophilia commented on src/components/ReportConfirmModal/index.tsx:
```diff
@@ -0,0 +1,50 @@
+}
+
+const ReportConfirmModal = ({ isOpen, onClose, onReport }: Props) => {
+  const handleReport = () => {
```
이것도 컨벤션에 맞지 않은 핸들러 함수 같아요! `handle` + `event` + `대상`의 형식을 맞춰주시면 좋을 것 같습니다!
`handleClickReportButton` 같은 이름이 적절할 것 같아요 : )

HY219 replied:
수정하겠습니다! 
DeleteConfirmModal 컴포넌트와 형식을 동일하게 가져갔는데, 추후에 이 컴포넌트도 수정하면 좋을 것 같네요! :)

HY219 replied:
리뷰 주신 것처럼 수정하였고, 기존에 정의돼 있던`handleClickReportButton` 이름의 함수는 역할에 맞게 `handleOpenReportConfirmModal`로 변경하였습니다:)

thalassophilia replied:
아직 handleReport 그대로인것 같은데 푸시 아직 안하신거겠죠?

HY219 replied:
푸쉬를 안했었네요..🤣 반영했습니다!!

thalassophilia commented on src/components/ReportConfirmModal/index.tsx:
```diff
@@ -0,0 +1,50 @@
+  const handleReport = () => {
+    onReport();
+    onClose();
+  };
+  return (
```
줄바꿈 해주세요!

thalassophilia commented on src/components/ReportConfirmModal/index.tsx:
```diff
@@ -0,0 +1,50 @@
+  };
+  return (
+    <Modal isOpen={isOpen} onClose={onClose} className="p-40pxr">
+      <Modal.Header hasCloseButton={true}> </Modal.Header>
```
저희 모달에서 Header가 필요한가에 대한 의문이 들기도 하네요ㅠ 헤더의 역할이 지금은 단순 x 버튼을 넣어주냐 마냐의 역할만 하고 있어서요! 혜연님이 구현해주신 부분이 아니긴 하지만 나중에 수정되면 좋을 부분이네요!

thalassophilia commented on src/components/ImageDetailModal/ImageMenuBar.tsx:
```diff
@@ -8,11 +10,23 @@ const ImageMenuBar = () => {
+      <ReportConfirmModal
+        isOpen={isOpen}
+        onClose={close}
+        onReport={() => "짤 이미지 신고 api 연결"} // TODO: [2024-03-03] 짤 이미지 신고 api 연결 - onReport={handleClickReportConfirm(imageId)}
```
이런 경우에 문자열을 반환하기보다 그냥 빈 함수로 두는게 더 좋을 것 같습니다!

HY219 committed:
Modify: ModalHeader 컴포넌트의 children Props를 옵셔널로 변경

전달한 children props가 없을 경우 Modal.Header를 좀 더 간결하게 작성

HY219 committed:
Rename: 컨벤션과 역할에 맞게 이벤트 핸들러 함수 이름 수정

handleClickReport -> handleOpenReportConfirmModal
handleReport -> handleClickReport
줄 간격 띄움

HY219 committed:
Refactor: hasCloseButton props에서 true 값 삭제

hasCloseButton props의 값은 동적으로 변경되지 않고 고정값 true를 가집니다.

HY219 committed:
Merge branch 'main' of https://github.com/zzalmyu/zzalmyu-frontend into #87/feature/report-confirmation-modal-ui

thalassophilia commented on src/components/ImageDetailModal/ImageMenuBar.tsx:
```diff
@@ -12,12 +12,12 @@ const ImageMenuBar = () => {
   const reportConfirmOverlay = useOverlay();
 
-  const handleClickReportButton = () => {
+  const handleOpenReportConfirmModal = () => {
```
다시 생각해보니 이 이름이 맞나에 대한 고민이 되네요ㅠ 어쨋든 '신고하기' 버튼을 클릭했을때 호출되는 핸들러 함수인데, 신고하기 버튼과는 좀 동떨어진, 버튼을 클릭했을때 발생하는 동작을 설명해주는 이름에 가깝다는 생각이 들어서요ㅠ
여기서도 handleClickReportButton이 맞는거 아닌가 싶은데 어떻게 생각하시나요?

HY219 replied:
`src/components/ReportConfirmModal/index.tsx` 파일에서 동일한 이름을 가진 이벤트 핸들러가 있어서 구분을 위해서 `handleOpenReportConfirmModal`로 작성해주었는데, 좋은 변수명이 떠오르지 않네요ㅜ

두 버튼 모두 신고를 위한 버튼이지만 다른 동작을 하는 버튼인데 동일한 이름을 사용해도 괜찮나요?

HY219 replied:
더 좋은 방법이 떠오르지 않아, 말씀해 주신 내용 반영했습니다!
핸들러 함수 이름에는 말씀해주신 것처럼 `handleClickReportButton`이 더 적합한 것 같네요:)

thalassophilia replied:
다른 컴포넌트이고, 둘다 신고 버튼을 눌렀을때 호출되는 함수라서 이름이 같아도 될 거 같아서요!

HY219 committed:
Refactor: import 및 hook 호출 순서 변경

useState를 useOverlay 앞에 배치
코드의 가독성 향상 및 일관성 유지

HY219 committed:
Rename: 짤 이미지 상세모달에서 신고 버튼 클릭 시 호출되는 이벤트 핸들러 함수 이름 변경

handleOpenReportConfirmModal -> handleClickReportButton
어떤 동작을 하는 함수인지에 대한 설명이 아닌, 클릭했을 때 발생하는 동작을 설명해주는 이름이므로 변경

HY219 committed:
Merge branch 'main' of https://github.com/zzalmyu/zzalmyu-frontend into #87/feature/report-confirmation-modal-ui

thalassophilia reviewed (approved):
LGTM!👍

choi-ik commented on src/components/ImageDetailModal/ImageMenuBar.tsx:
```diff
@@ -1,18 +1,31 @@
   const [isLiked, setIsLiked] = useState(false);
+  const reportConfirmOverlay = useOverlay();
 
   const handleClickLike = () => {
```
이 버튼 토글 버튼인 것 같은데 handleClickLikeToggle은 아주 별로인 네이밍인것 같지만 이런 느낌 비슷하게 toggle 기능을 한다는 네이밍이 들어가는 게 좋지 않을까 싶어요!

HY219 replied:
이 부분은 제가 작업한 부분이 아닌데, 현재 PR에서 작업하는게 맞을까요? 좋아요와 관련없는 PR이여서요!

choi-ik commented on src/components/ReportConfirmModal/index.tsx:
```diff
@@ -0,0 +1,51 @@
+      <Modal.Header hasCloseButton />
+      <Modal.Body>
+        <div className="flex flex-col items-center">
+          <AlertTriangle color="#ED0000" strokeWidth="1.2" size="84" />
```
이 부분 aria-label 추가해 주셔야 할 것 같습니다 !

HY219 replied:
AlertTriangle 이라는 이름으로 아이콘 이름이 있어서 aria-label을 따로 추가해 주진 않았어요!

이전에 구현되어있던 `DeleteConfirmModal.tsx` 파일의 코드를 그대로 가져온 부분인데, aria-label를 `DeleteConfirmModal.tsx` 파일과 현재 파일에 모두 추가해준다면 더욱 명시적으로 보일 수는 있겠네요:)

HY219 committed:
Refactor: AlertTrigangle에 aria-label 추가

경고삼각형 아이콘

HY219 committed:
Merge branch 'main' of https://github.com/zzalmyu/zzalmyu-frontend into #87/feature/report-confirmation-modal-ui

yjc2021 reviewed (approved):
LGTM입니다 👍
